### PR TITLE
fix: select correct window in react-image-crop-picker when input window is present

### PR DESCRIPTION
**Pull request by brainbicycle**
The type of this PR is: **Bugfix**

This PR resolves [CX-711]

### Description

We do some swizzling in the AppDelegate that causes  `[[UIApplication sharedApplication] delegate] window]` to return nil. react-native-image-crop-picker uses this method to find the viewController to present the picker onto so we had a patch that grabbed the last window in the hierarchy to workaround this issue. In the My Collection flow this was returning the wrong window, specifically a UIInputWindow that manages presentation of the keyboard. This PR updates our existing patch for react-native-image-crop-picker to look for our custom window subclass instead which should return the correct window, and falls back to the last window if it can't find the subclass.

#### Before 

before-image-crop

#### After 

after-image-crop

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-711]: https://artsyproduct.atlassian.net/browse/CX-711

### CODE REVIEW

**Comment by damassi** [comment]
So happy to see this fixed! Ty 👍

File: patches/react-native-image-crop-picker+0.32.2.patch [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
++        if ([window isKindOfClass:NSClassFromString(@"ARWindow")]) {
++            presentingWindow = window;
++        }
++    }
++    if (!presentingWindow) {
++        NSAssert(false, @"Unable to find ARWindow did it get renamed?");
++        presentingWindow = [[[UIApplication sharedApplication] windows] lastObject];
++    }
++
++    UIViewController *root = [presentingWindow rootViewController];
```
Comment (damassi) [inline patches/react-native-image-crop-picker+0.32.2.patch:26]: @ashfurrow - did you make some kind of similar change recently? This vaguely brings to mind another bug we encountered, but I can't remember what.

**Review by ds300 — APPROVED** [review]
nice one! TIL about NSClassFromString!

File: patches/react-native-image-crop-picker+0.32.2.patch (hunk 1 above)
Comment (ashfurrow) [reply patches/react-native-image-crop-picker+0.32.2.patch:26]: Yeah, it was to fix this same bug actually! But that change didn't cover the case that Brian outlined, this patch is much more specific 👍

File: patches/react-native-image-crop-picker+0.32.2.patch [hunk 2]
```diff
@@ -1,17 +1,29 @@
...
-+    UIViewController *root = [[[[UIApplication sharedApplication] windows] lastObject] rootViewController];
++- (UIViewController *) getRootVC {
++    UIWindow *presentingWindow;
++    for (UIWindow *window in [[UIApplication sharedApplication] windows]) {
++        if ([window isKindOfClass:NSClassFromString(@"ARWindow")]) {
++            presentingWindow = window;
++        }
++    }
++    if (!presentingWindow) {
++        NSAssert(false, @"Unable to find ARWindow did it get renamed?");
```
Comment (ashfurrow) [inline patches/react-native-image-crop-picker+0.32.2.patch:22]: Great work, this 👍

**Review by ashfurrow — APPROVED** [review]
🙌

**Pull request merged**